src/realm/object-store/shared_realm.cpp

```diff
@@ -822,6 +822,10 @@ bool Realm::do_refresh()
         return false;
     }
 
+    if (m_config.immutable()) {
+        throw std::logic_error("Can't refresh a read-only Realm.");
```

Comment:
Can we instead return `false` like we do for frozen Realms?

Reply:
Cocoa throws an exception if the user explicitly calls refresh on an immutable Realm because it's almost certainly a sign that the developer is trying to do something that won't/can't work and we want to let them know. It makes it awkward if they write a function which accepts an arbitrary Realm and they want to call refresh() on it, but that's a terrible idea so making it awkward isn't necessarily a bad thing.

Reply:
Makes sense, but isn't it the case for frozen realms as well? I guess in my mind those two are very similar and it just triggers my OCD a little that we have different behavior here.

Reply:
Allowing calling refresh() on frozen Realms does seem like a mistake to me.